File: server/internal/model/team.go
package model

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/datatypes"
)

// Team 团队模型
type Team struct {
	Model
	WorkspaceID     uuid.UUID      `gorm:"type:uuid;not null;index" json:"workspace_id"`
	ParentID        *uuid.UUID     `gorm:"type:uuid;index" json:"parent_id,omitempty"`
	Name            string         `gorm:"type:varchar(255);not null" json:"name"`
	Key             string         `gorm:"type:varchar(10);uniqueIndex;not null" json:"key"`
	IconURL         *string        `gorm:"type:text" json:"icon_url,omitempty"`
	Timezone        string         `gorm:"type:varchar(64);not null;default:'UTC'" json:"timezone"`
	IsPrivate       bool           `gorm:"not null;default:false" json:"is_private"`
	CycleSettings   datatypes.JSON `gorm:"type:jsonb;default:'{}'" json:"cycle_settings,omitempty"`
	WorkflowSettings datatypes.JSON `gorm:"type:jsonb;default:'{}'" json:"workflow_settings,omitempty"`

	// 关联关系
	Workspace    *Workspace      `gorm:"foreignKey:WorkspaceID;constraint:OnDelete:CASCADE" json:"workspace,omitempty"`
	Parent       *Team           `gorm:"foreignKey:ParentID;constraint:OnDelete:SET NULL" json:"parent,omitempty"`
	Children     []Team          `gorm:"foreignKey:ParentID;constraint:OnDelete:SET NULL" json:"children,omitempty"`
	Members      []TeamMember    `gorm:"foreignKey:TeamID;constraint:OnDelete:CASCADE" json:"members,omitempty"`
	Issues       []Issue         `gorm:"foreignKey:TeamID;constraint:OnDelete:CASCADE" json:"issues,omitempty"`
	WorkflowStates []WorkflowState `gorm:"foreignKey:TeamID;constraint:OnDelete:CASCADE" json:"workflow_states,omitempty"`
	Cycles       []Cycle         `gorm:"foreignKey:TeamID;constraint:OnDelete:CASCADE" json:"cycles,omitempty"`
}

// TableName 指定表名
func (Team) TableName() string {
	return "teams"
}

// TeamMember 团队成员模型
type TeamMember struct {
	TeamID   uuid.UUID `gorm:"type:uuid;primaryKey;not null" json:"team_id"`
	UserID   uuid.UUID `gorm:"type:uuid;primaryKey;not null" json:"user_id"`
	Role     Role      `gorm:"type:varchar(20);not null;default:'member'" json:"role"`
	JoinedAt time.Time `gorm:"not null;default:now()" json:"joined_at"`

	// 关联关系
	Team *Team `gorm:"foreignKey:TeamID;constraint:OnDelete:CASCADE" json:"team,omitempty"`
	User *User `gorm:"foreignKey:UserID;constraint:OnDelete:CASCADE" json:"user,omitempty"`
}

// TableName 指定表名
func (TeamMember) TableName() string {
	return "team_members"
}

File: web/src/types/team.ts
/**
 * Team 和 TeamMember 相关类型定义
 */

// 团队角色
export type TeamRole = 'admin' | 'member';

// 团队信息
export interface Team {
  id: string;
  workspace_id: string;
  name: string;
  key: string;
  description: string;
  created_at: string;
  updated_at: string;
}

// 团队成员
export interface TeamMember {
  id: string; // user_id
  user_id: string;
  role: TeamRole;
  joined_at: string;
  user?: {
    id: string;
    name: string;
    email: string;
    username: string;
    avatar_url?: string;
  };
}

// 团队列表响应
export interface TeamListResponse {
  teams: Team[];
  total: number;
  page: number;
}

// 创建团队请求
export interface CreateTeamRequest {
  name: string;
  key: string;
  description: string;
  workspace_id: string;
}

// 更新团队请求
export interface UpdateTeamRequest {
  name?: string;
  key?: string;
  description?: string;
}

// 添加成员请求
export interface AddMemberRequest {
  user_id: string;
  role: TeamRole;
}

// 成员列表响应
export interface MemberListResponse {
  members: TeamMember[];
}
